src/Symfony/Component/DependencyInjection/Compiler/RegisterServiceSubscribersPass.php

```diff
@@ -85,6 +87,11 @@ protected function processValue($value, bool $isRoot = false)
                 if (!$autowire) {
                     throw new InvalidArgumentException(sprintf('Service "%s" misses a "container.service_subscriber" tag with "key"/"id" attributes corresponding to entry "%s" as returned by "%s::getSubscribedServices()".', $this->currentId, $key, $class));
                 }
+                if ($replaceDeprecatedSession && SessionInterface::class === $type) {
+                    // This prevents triggering the deprecation when building the container
```

Comment:
Likely stupid: could #39821 help here?

Reply:
This code prevents triggering a deprecation when the service/alias `SessionInterface` is declared in the AbstractController ServiceLocator (whether or not it's used).

#39821 prevents triggering a deprecation for the alias when the linked service is also deprecated.